src/test/java/com/redhat/devtools/intellij/tektoncd/utils/DeployHelperTest.java

```diff
@@ -0,0 +1,164 @@
+import org.junit.Test;
+import org.mockito.MockedStatic;
+
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class DeployHelperTest extends FixtureBaseTest {
+
+    private static final String RESOURCE_PATH = "utils/deployhelper/";
+    private String pipeline_yaml;
+    private TknCli tkn;
+
+    @Before
+    public void setUp() throws Exception {
+        super.setUp();
+
+        tkn = mock(TknCli.class);
+        pipeline_yaml = load(RESOURCE_PATH + "pipeline.yaml");
+    }
+
+
+    @Test
+    public void SaveOnCluster_InvalidYamlMissingKind_Throw() throws IOException {
+        String yaml = load(RESOURCE_PATH + "missingkind_pipeline.yaml");
+        assertIsCorrectErrorWhenInvalidYaml(yaml, "Tekton file has not a valid format. Kind field is not found.");
+    }
+
+    @Test
+    public void SaveOnCluster_InvalidYamlMissingName_Throw() throws IOException {
+        String yaml = load(RESOURCE_PATH + "missingname_pipeline.yaml");
+        assertIsCorrectErrorWhenInvalidYaml(yaml, "Tekton file has not a valid format. Name field is not valid or found.");
+    }
+
+    @Test
+    public void SaveOnCluster_InvalidYamlMissingApiVersion_Throw() throws IOException {
+        String yaml = load(RESOURCE_PATH + "missingapiversion_pipeline.yaml");
+        assertIsCorrectErrorWhenInvalidYaml(yaml, "Tekton file has not a valid format. ApiVersion field is not found.");
+    }
+
+    @Test
+    public void SaveOnCluster_InvalidYamlInvalidApiVersion_Throw() throws IOException {
+        String yaml = load(RESOURCE_PATH + "missingcrd_pipeline.yaml");
+        assertIsCorrectErrorWhenInvalidYaml(yaml, "Tekton file has not a valid format. ApiVersion field contains an invalid value.");
+    }
+
+    private void assertIsCorrectErrorWhenInvalidYaml(String yaml, String expectedError) {
+        try (MockedStatic<TreeHelper> theMock = mockStatic(TreeHelper.class)) {
+            theMock.when(() -> TreeHelper.getTkn(any())).thenReturn(null);
+            try {
+                DeployHelper.saveOnCluster(null, yaml, "", false, true);
+            } catch (IOException e) {
+                assertEquals(expectedError, e.getLocalizedMessage());
+            }
+        }
+    }
+
+    @Test
+    public void SaveOnCluster_SkipConfirmationIsTrue_IsSaveConfirmedNotCalled() {
+        try(MockedStatic<UIHelper> uiHelperMockedStatic = mockStatic(UIHelper.class)) {
+            DeployHelper.saveOnCluster(null, pipeline_yaml, true);
+            uiHelperMockedStatic.verify(times(0), () -> UIHelper.executeInUI(any(Runnable.class)));
+        } catch (IOException e) { }
```

Comment:
Any reason why IOException being raised does not fail tests

Reply:
Why should it fail? If an exception is raised it goes inside the catch and nothing is done there. I'm just verifying that the dialog is not called if the skip flag is set to true. The code should run fine until that point bc i'm feeding the save method with a working pipeline, after that the code can also fails, i don't care.